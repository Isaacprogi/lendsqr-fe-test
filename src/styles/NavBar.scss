@import '../styles/variables/mixins';
@import '../styles/variables/variables';

.nav-bar{
    width: 100%;
    height: 100px;
    display: flex; 
    align-items: center;
    justify-content: space-between;
    background:$whiteColor;
    box-shadow: 10px 5px 20px rgba(0, 0, 0, 0.04);
    position: sticky;
    top: 0;
    z-index: 900;

    .logo-search-section{
     display: flex;
     align-items: center;
     justify-content: flex-start;
     width: 100%;
     margin-left: 15px;



     .search{
         display: flex;
         align-items: center;
         justify-content: flex-start;
         width: 90%;
         height: 40px;
         background:$whiteColor;
         border: 1px solid #a0acc5;
         border-radius: 8px;
         overflow: hidden;
         margin-right: 1rem;
         padding-left: 1rem;

         @include largeDevices {
            margin-left: 6rem;  
            max-width: 400px; 
        }
         

         input{
             outline: none;
             border: none;
             height: 100%;
             width: 100%;

         }
         
         

         .search-icons-box{
             width: 56px;
             background: #39CDCC;
             display: flex;
             align-items: center;
             justify-content: center;
             height: 100%;
             cursor: pointer;


             .search-icon{
                 color: $whiteColor;
             }
         }

         
     }

     .logo{
         display: flex;
         align-items: center;
         justify-content: flex-start;
         margin-right: 1rem;

         .lendsqr{
            height: 30px;
            display: none;
            margin-left: 1rem;
            
            @include largeDevices {
                display: block;      
            }
        }
        
     }



    }


    .others{
        display: none;
        align-items: center;
        justify-content: space-between;
        margin-right:2rem;

        @include largeDevices{
            margin-right: 4rem;
        }

    
     
        img{
            width: 48px;
            height:48px;
            border-radius: 50%;
            overflow:hidden;
        }


        div{
            display: flex;
            align-items: center;
            justify-content: space-between;

            span{
                color: #213F7D;
                font-family: 'Work Sans';
                font-style: normal;
                font-weight: 500;
                font-size: 16px;
            }
        }

        .docs,.notification-icon,div{
            margin-right: 2rem;
        }
        .avatar{
           margin-right: .8rem;
        }

        .notification-icon{
            width: 30px;
            height: 30px;
            cursor: pointer;
            border-radius: 100%;
            padding: .2rem;

            &:hover{
                background-color: rgb(228, 219, 219);
                transition: all 0.3s ease;
            }
        }

        .docs{
            text-decoration: underline;
            cursor: pointer;
            font-style: normal;
            font-weight: 400;
            font-size: 16px;
            color: #213F7D;
        }

        .drop-down-icon{
            color: #213F7D;
            cursor: pointer;
        }

        @include mediumDevices {
            display: flex;     
        }

        
    }

    .fa-times{
        color:#009F95;
        border: none;
        position: absolute;
        right: 1rem;
        top: 1rem;
        outline: none;
    }


    .fa-bars{
         margin-right: 15px;
         cursor: pointer;
         border-radius: 100%;
         display: flex;

         img{
            width: 100%;
            height: 100%;
         }

         .hidden{
             display: none;
         }
    


        @include mediumDevices {
            display: none;     
        }
    }

    .fixed-element {
        position: fixed;
        top: 0;
        left: 0;
        width: 100%; 
        max-width: 18rem;
        height: 100%; 
        background-color: white;
        box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); 
        z-index: 20;
        overflow-y: auto;
      
        @media (min-width: 768px) {
          display: none; 
        }
      }
      

    .right-drawer{
        position: fixed;
        width: 15rem;
        background-color: rgb(17, 26, 49);
        height: max-content;
        top: 0;
        right: 0;
        border-left: .1px solid #dddfe6;
        border-bottom-left-radius: 10px;
        transition: all .3s;
        transform: scale(0) rotate(180deg);
        transform-origin: top right;
        padding-bottom: 1rem;
        z-index: 1000;
        box-shadow: 10px 5px 20px rgba(0, 0, 0, 0.04);

        .cancel{
            position: absolute;
            right: 20px;
            top: 20px;
            color: rgb(233, 220, 220);
            cursor: pointer;
        }

        .profile-top{
            display: flex;
            width: 100%;
            align-items: center;
            justify-content: flex-start;
            margin-top: 2rem;
            color: $whiteColor;
            

            .avatar{
               width: 3.5rem;
               height: 3.5rem;
               border-radius: 50%;
               margin-left: 1rem;
               margin-right: .5rem;
               overflow: hidden;
               
               img{
                   width: 100%;
                   height: 100%;
               }
            }
            .others{
                span{
                    color: white;
                }
            }
        }

        .drawer-list{
            display: flex;
            flex-direction: column;
            align-items: flex-start;
            justify-content: center;
            margin-top:1rem;
            
            
            >span{
                width: 100%;
                display: flex;
                align-items: center;
                padding: .5rem;
                cursor: pointer;
                color: white;

                .drawer-list-icons{
                    font-size: 1.5rem;
                    margin-right: .3rem;
                }

                &:hover{
                    background-color: #848fa2;
                }
                
            }
        }

        @include mediumDevices {
            display: none;     
        }
    
    }

    .visible{
        transform: scale(1);
        
    }

}

