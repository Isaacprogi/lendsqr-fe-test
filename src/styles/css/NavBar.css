.nav-bar {
  width: 100%;
  height: 100px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  background: white;
  -webkit-box-shadow: 10px 5px 20px rgba(0, 0, 0, 0.04);
          box-shadow: 10px 5px 20px rgba(0, 0, 0, 0.04);
  position: sticky;
  top: 0;
  z-index: 900;
}
.nav-bar .logo-search-section {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  width: 100%;
  margin-left: 15px;
}
.nav-bar .logo-search-section .search {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  width: 90%;
  height: 40px;
  background: white;
  border: 1px solid #a0acc5;
  border-radius: 8px;
  overflow: hidden;
  margin-right: 1rem;
  padding-left: 1rem;
}
@media screen and (min-width: 988px) {
  .nav-bar .logo-search-section .search {
    margin-left: 6rem;
    max-width: 400px;
  }
}
.nav-bar .logo-search-section .search input {
  outline: none;
  border: none;
  height: 100%;
  width: 100%;
}
.nav-bar .logo-search-section .search .search-icons-box {
  width: 56px;
  background: #39CDCC;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  height: 100%;
  cursor: pointer;
}
.nav-bar .logo-search-section .search .search-icons-box .search-icon {
  color: white;
}
.nav-bar .logo-search-section .logo {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  margin-right: 1rem;
}
.nav-bar .logo-search-section .logo .lendsqr {
  height: 30px;
  display: none;
  margin-left: 1rem;
}
@media screen and (min-width: 988px) {
  .nav-bar .logo-search-section .logo .lendsqr {
    display: block;
  }
}
.nav-bar .others {
  display: none;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  margin-right: 2rem;
}
@media screen and (min-width: 988px) {
  .nav-bar .others {
    margin-right: 4rem;
  }
}
.nav-bar .others img {
  width: 48px;
  height: 48px;
  border-radius: 50%;
  overflow: hidden;
}
.nav-bar .others div {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
}
.nav-bar .others div span {
  color: #213F7D;
  font-family: "Work Sans";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
}
.nav-bar .others .docs, .nav-bar .others .notification-icon, .nav-bar .others div {
  margin-right: 2rem;
}
.nav-bar .others .avatar {
  margin-right: 0.8rem;
}
.nav-bar .others .notification-icon {
  width: 30px;
  height: 30px;
  cursor: pointer;
  border-radius: 100%;
  padding: 0.2rem;
}
.nav-bar .others .notification-icon:hover {
  background-color: rgb(228, 219, 219);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}
.nav-bar .others .docs {
  text-decoration: underline;
  cursor: pointer;
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  color: #213F7D;
}
.nav-bar .others .drop-down-icon {
  color: #213F7D;
  cursor: pointer;
}
@media screen and (min-width: 768px) {
  .nav-bar .others {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
  }
}
.nav-bar .fa-times {
  color: #009F95;
  border: none;
  position: absolute;
  right: 1rem;
  top: 1rem;
  outline: none;
}
.nav-bar .fa-bars {
  margin-right: 15px;
  cursor: pointer;
  border-radius: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}
.nav-bar .fa-bars img {
  width: 100%;
  height: 100%;
}
.nav-bar .fa-bars .hidden {
  display: none;
}
@media screen and (min-width: 768px) {
  .nav-bar .fa-bars {
    display: none;
  }
}
.nav-bar .fixed-element {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  max-width: 18rem;
  height: 100%;
  background-color: white;
  -webkit-box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
          box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  z-index: 20;
  overflow-y: auto;
}
@media (min-width: 768px) {
  .nav-bar .fixed-element {
    display: none;
  }
}
.nav-bar .right-drawer {
  position: fixed;
  width: 15rem;
  background-color: rgb(17, 26, 49);
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  top: 0;
  right: 0;
  border-left: 0.1px solid #dddfe6;
  border-bottom-left-radius: 10px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transform: scale(0) rotate(180deg);
          transform: scale(0) rotate(180deg);
  -webkit-transform-origin: top right;
          transform-origin: top right;
  padding-bottom: 1rem;
  z-index: 1000;
  -webkit-box-shadow: 10px 5px 20px rgba(0, 0, 0, 0.04);
          box-shadow: 10px 5px 20px rgba(0, 0, 0, 0.04);
}
.nav-bar .right-drawer .cancel {
  position: absolute;
  right: 20px;
  top: 20px;
  color: rgb(233, 220, 220);
  cursor: pointer;
}
.nav-bar .right-drawer .profile-top {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  margin-top: 2rem;
  color: white;
}
.nav-bar .right-drawer .profile-top .avatar {
  width: 3.5rem;
  height: 3.5rem;
  border-radius: 50%;
  margin-left: 1rem;
  margin-right: 0.5rem;
  overflow: hidden;
}
.nav-bar .right-drawer .profile-top .avatar img {
  width: 100%;
  height: 100%;
}
.nav-bar .right-drawer .profile-top .others span {
  color: white;
}
.nav-bar .right-drawer .drawer-list {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: start;
      -ms-flex-align: start;
          align-items: flex-start;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  margin-top: 1rem;
}
.nav-bar .right-drawer .drawer-list > span {
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding: 0.5rem;
  cursor: pointer;
  color: white;
}
.nav-bar .right-drawer .drawer-list > span .drawer-list-icons {
  font-size: 1.5rem;
  margin-right: 0.3rem;
}
.nav-bar .right-drawer .drawer-list > span:hover {
  background-color: #848fa2;
}
@media screen and (min-width: 768px) {
  .nav-bar .right-drawer {
    display: none;
  }
}
.nav-bar .visible {
  -webkit-transform: scale(1);
          transform: scale(1);
}/*# sourceMappingURL=NavBar.css.map */