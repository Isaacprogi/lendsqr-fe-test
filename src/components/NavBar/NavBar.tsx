import '../../styles/css/NavBar.css';
import { useState } from 'react';
import UNION from '../../assets/svg/login/union.svg';
import LENDSQR from '../../assets/svg/login/lendsqr.svg';
import { FaSearch, FaTimes } from 'react-icons/fa';
import { IoMdNotificationsOutline, IoMdArrowDropdown } from 'react-icons/io';
import Me from '../../assets/images/Profile.png';
import { FaBars } from 'react-icons/fa';
import { SideBar } from '../SideBar/SideBar';
import { motion } from 'framer-motion';

export const NavBar = () => {
  const [displayDrawer, setDisplayDrawer] = useState(false);

  const handleDisplayDrawer = () => {
    setDisplayDrawer(!displayDrawer);
  };

  const sidebarVariants = {
    open: { x: 0 },
    closed: { x: '-100%' },
  };

  const [isOpen, setIsOpen] = useState(false);

  const toggleMenu = () => {
    setIsOpen(!isOpen);
  };

  return (
    <div className='nav-bar'>
      <div className='logo-search-section'>
        <div className='logo'>
          <img className='union' src={UNION} alt="union-icon" />
          <img className='lendsqr' src={LENDSQR} alt="lendsqr-icon" />
        </div>
        <div className='search'>
          <input type="text" placeholder='search for anything' />
          <span className='search-icons-box'>
            <FaSearch data-testid="search-icon" className='search-icon' />
          </span>
        </div>
      </div>
      <div className='others'>
        <span data-testid="docs-span" className='docs'>Docs</span>
        <IoMdNotificationsOutline data-testid="notification-icon" className='notification-icon' />
        <img src={Me} alt="avatar" className='avatar' />
        <div>
          <span data-testid='user-name-1'>Adeji</span>
          <IoMdArrowDropdown data-testid="arrow-down-icon" className='drop-down-icon' />
        </div>
      </div>
      <span data-testid='fa-bars' onClick={handleDisplayDrawer} className='fa-bars'>
        <FaBars onClick={toggleMenu} />
      </span>
      <motion.div
        initial="closed"
        animate={isOpen ? "open" : "closed"}
        variants={sidebarVariants}
        transition={{ type: 'spring', stiffness: 300, damping: 30 }}
        className="fixed-element"
      >
        <div>
          <FaTimes data-testid="fa-times" className="fa-times" onClick={toggleMenu} size={24} />
          <SideBar />
        </div>
      </motion.div>
    </div>
  );
};
