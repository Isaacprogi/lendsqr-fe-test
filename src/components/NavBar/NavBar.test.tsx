import { fireEvent, render, screen } from '@testing-library/react';
import { NavBar } from './NavBar';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';  // Import MemoryRouter

describe('NavBar Component', () => {
    test('top right drawer should function perfectly', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );

        const FaBarsElement = screen.getByTestId('fa-bars');
        const SearchIconElement = screen.getByTestId('search-icon');
        const DocsSpanElement = screen.getByTestId('docs-span');
        const NotificationIconElement = screen.getByTestId('notification-icon');
        const UserNameElement = screen.getByTestId('user-name-1');
        const ArrowDownIconElement = screen.getByTestId('arrow-down-icon');

        expect(SearchIconElement).toBeInTheDocument();
        expect(DocsSpanElement).toBeInTheDocument();
        expect(NotificationIconElement).toBeInTheDocument();
        expect(UserNameElement).toBeInTheDocument();
        expect(ArrowDownIconElement).toBeInTheDocument();

        fireEvent.click(FaBarsElement);
        const FaTimesElement = screen.getByTestId('fa-times');
        expect(FaTimesElement).toBeInTheDocument();

        fireEvent.click(FaTimesElement);
        expect(FaBarsElement).toBeInTheDocument();
    });

    test('should match snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
