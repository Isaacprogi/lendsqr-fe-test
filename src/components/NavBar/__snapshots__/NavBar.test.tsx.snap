// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Component > should match snapshot 1`] = `
<div
  className="nav-bar"
>
  <div
    className="logo-search-section"
  >
    <div
      className="logo"
    >
      <img
        alt="union-icon"
        className="union"
        src="/src/assets/svg/login/union.svg"
      />
      <img
        alt="lendsqr-icon"
        className="lendsqr"
        src="/src/assets/svg/login/lendsqr.svg"
      />
    </div>
    <div
      className="search"
    >
      <input
        placeholder="search for anything"
        type="text"
      />
      <span
        className="search-icons-box"
      >
        <svg
          className="search-icon"
          data-testid="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="others"
  >
    <span
      className="docs"
      data-testid="docs-span"
    >
      Docs
    </span>
    <svg
      className="notification-icon"
      data-testid="notification-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257 120.471c7.083 0 23.911 4.479 23.911 4.479 45.589 10.447 77.678 52.439 77.678 99.85V352.412l9.321 9.364 7.788 7.823H136.302l7.788-7.823 9.321-9.364V224.8c0-47.41 32.089-89.403 77.678-99.85 0 0 18.043-4.479 23.911-4.479M256 48c-17.602 0-31.059 13.518-31.059 31.2v14.559c-59.015 13.523-103.53 67.601-103.53 131.041v114.4L80 380.8v20.8h352v-20.8l-41.411-41.6V224.8c0-63.44-44.516-117.518-103.53-131.041V79.2c0-17.682-13.457-31.2-31.059-31.2zm41.411 374.4h-82.823c0 22.881 18.633 41.6 41.412 41.6s41.411-18.719 41.411-41.6z"
      />
    </svg>
    <img
      alt="avatar"
      className="avatar"
      src="/src/assets/images/Profile.png"
    />
    <div>
      <span
        data-testid="user-name-1"
      >
        Adeji
      </span>
      <svg
        className="drop-down-icon"
        data-testid="arrow-down-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 192l128 128 128-128z"
        />
      </svg>
    </div>
  </div>
  <span
    className="fa-bars"
    data-testid="fa-bars"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
  </span>
  <div
    className="fixed-element"
    style={
      {
        "transform": "translateX(-100%) translateZ(0)",
      }
    }
  >
    <div>
      <svg
        className="fa-times"
        data-testid="fa-times"
        fill="currentColor"
        height={24}
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
      <div
        className="sidebar"
      >
        <section
          className="profile-more-options"
          data-testid="profile-more-options"
        >
          <div
            className="profile-info"
          >
            <span
              className="avatar"
            >
              <img
                alt=""
                data-testid="avatar"
                src="/src/assets/images/Profile.png"
              />
            </span>
            <span
              data-testid="user-name-2"
            >
              Adeji
            </span>
          </div>
          <a
            href="/"
            onClick={[Function]}
          >
            <div
              className="top-bottom-icon-duo"
            >
              <svg
                data-testid="notification-drawer-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M816 768h-24V428c0-141.1-104.3-257.8-240-277.2V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.8C336.3 170.2 232 286.9 232 428v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48z"
                />
              </svg>
            </div>
            <span>
               Notifications
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <div
              className="top-bottom-icon-duo"
            >
              <svg
                aria-hidden="true"
                data-testid="docs-drawer-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M4 4a2 2 0 012-2h4.586A2 2 0 0112 2.586L15.414 6A2 2 0 0116 7.414V16a2 2 0 01-2 2H6a2 2 0 01-2-2V4z"
                  fillRule="evenodd"
                />
              </svg>
            </div>
            <span>
              Docs
            </span>
          </a>
        </section>
        <section
          className="top"
        >
          <a
            className="org"
            href="/"
            onClick={[Function]}
          >
            <div
              className="top-bottom-icon-duo"
            >
              <img
                alt="switch-organization-icon-1"
                src="/src/assets/svg/sidebar/OrganizationTop.svg"
              />
              <img
                alt="switch-organization-icon-2"
                src="/src/assets/svg/sidebar/OrganizationBottom.svg"
              />
            </div>
            <span>
              Switch Organization
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
              />
            </svg>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <div
              className="top-bottom-icon-duo"
            >
              <img
                alt="dashboard-icons-1"
                src="/src/assets/svg/sidebar/DashBoardTop.svg"
              />
              <img
                alt="dashboard-icon-2"
                className="dash-board-bottom"
                src="/src/assets/svg/sidebar/DashBoardBottom.svg"
              />
            </div>
            <span>
              Dash Board
            </span>
          </a>
        </section>
        <section
          className="middle"
        >
          <h3>
            Customers
          </h3>
          <a
            href="/users"
            onClick={[Function]}
          >
            <img
              alt="users-icon"
              src="/src/assets/svg/sidebar/Users.svg"
            />
            <span>
              Users
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="guarantors-icon"
              src="/src/assets/svg/sidebar/Guarantors.svg"
            />
            <span>
              Guarantors
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="loans-icon"
              src="/src/assets/svg/sidebar/Loans.svg"
            />
            <span>
              Loans
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="decision-icon"
              src="/src/assets/svg/sidebar/Decision.svg"
            />
            <span>
              Decision Model
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="savings-icon"
              src="/src/assets/svg/sidebar/Savings.svg"
            />
            <span>
              Savings
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="loan-request-icon"
              src="/src/assets/svg/sidebar/LoanRequests.svg"
            />
            <span>
              Loan Requests
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="white-list-icon"
              src="/src/assets/svg/sidebar/WhiteList.svg"
            />
            <span>
              whiteList
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="karma-icon"
              src="/src/assets/svg/sidebar/Karma.svg"
            />
            <span>
              Karma
            </span>
          </a>
          <h3>
            Businesses
          </h3>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="organization-icon"
              src="/src/assets/svg/sidebar/briefcase.svg"
            />
            <span>
              Organization
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="loan-products-icon"
              src="/src/assets/svg/sidebar/Guarantors.svg"
            />
            <span>
              Loan Products
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="savings-product-icon"
              src="/src/assets/svg/sidebar/SavingsNew.svg"
            />
            <span>
              Savings Products
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="fees-and-charges-icon"
              src="/src/assets/svg/sidebar/fees.svg"
            />
            <span>
              Fees and Charges
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="transactions-icon"
              src="/src/assets/svg/sidebar/trans.svg"
            />
            <span>
              Transactions
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="services-icon"
              src="/src/assets/svg/sidebar/Services.svg"
            />
            <span>
              Services
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="service-amount-icon"
              src="/src/assets/svg/sidebar/ServiceAmount.svg"
            />
            <span>
              Service Account
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="settlements-icon"
              src="/src/assets/svg/sidebar/settlements.svg"
            />
            <span>
              Settlements
            </span>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="reports-icon"
              src="/src/assets/svg/sidebar/chart-bar.svg"
            />
            <span>
              Reports
            </span>
          </a>
          <span
            className="end-section"
          >
            <h3>
              Businesses
            </h3>
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="preferences-icon"
                src="/src/assets/svg/sidebar/Preferences.svg"
              />
              <span>
                Preferences
              </span>
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="fees-and-pricing-icon"
                src="/src/assets/svg/sidebar/pricing.svg"
              />
              <span>
                Fees and Pricing
              </span>
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="audit-logs-icon"
                src="/src/assets/svg/sidebar/auditLogs.svg"
              />
              <span>
                Audit Logs
              </span>
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="systen-messages-icon"
                src="/src/assets/svg/sidebar/sysmessages.svg"
              />
              <span>
                System Messages
              </span>
            </a>
          </span>
          <div
            className="log-out-section"
          >
            <span
              className="log-out"
              onClick={[Function]}
            >
              <img
                alt="log-out-icon"
                src="/src/assets/svg/sidebar/sign-out.svg"
              />
              <span>
                Logout
              </span>
            </span>
            <span
              className="v1"
            >
              v1.2.0
            </span>
          </div>
        </section>
        <section
          className="bottom"
        />
      </div>
    </div>
  </div>
</div>
`;
